### PR TITLE
[fix]: fix layer norm

### PR DESCRIPTION
**Pull request by and-rewsmith**
Three conceptual changes:
1. Ran some experiments on layer norm
2. Changed activation functions
3. Fixed hyperparam search
4. Sparse connections

*Layer Norm*
Layer normalization was not properly done, as it was not applied to the activations of the layer before. Fixing that.

*Changing activation functions*
Changing to elu / prelu seems to give better results

*Hyperparam search*
Fixing the script to not spam wandb if there is some fatal error.

**Sparse Connections**
Tried making forward and backward connections sparse, as it is biologically plausible, however this didn't show promising results. Therefore default density is 1.

Screenshot of training run showing increased perf:
image

### CODE REVIEW

File: config.toml [hunk 1]
```diff
@@ -1,14 +1,14 @@
 [model]
-hidden_sizes = [3000, 3000, 3000]
```
Comment (and-rewsmith) [inline config.toml:2]: The old file is technically a little better, but I like the smooth loss for latent softmax classifier. So I am making the change. It is only a little worse.
